### PR TITLE
Add support for plots on reference surfaces

### LINKED ISSUE #255: Fix lookup by "failure_label"
The enum FailureOutput is used to get values with a given failure label (MODE, VALUE or MAX_LAYER_INDEX). In pydpf-composites the enum maps to a hardcoded index whereas in the backend the index is determined by string support. This means that the indices potentially can become inconsistent.
String supports are not exposed in python, we probably need to look into a dpf DataTree to implement this.

### PR DESCRIPTION
**Pull request by janvonrickenbach**
Implement mapping to reference surface in composite model. The change is fully backward compatible. The evaluate_failure_criteria function just returns a container with more fields. 

Also the fields are not identified by failure_index anymore, but the field name is used which closes  #255

### CODE REVIEW

File: src/ansys/dpf/composites/composite_model.py [hunk 1]
```diff
@@ -147,6 +147,7 @@ def evaluate_failure_criteria(
         composite_scope: Optional[CompositeScope] = None,
         measure: FailureMeasureEnum = FailureMeasureEnum.INVERSE_RESERVE_FACTOR,
         write_data_for_full_element_scope: bool = True,
```
Comment (janvonrickenbach) [inline src/ansys/dpf/composites/composite_model.py:149]: Activate the max_chunk_size input which was ignored

File: .github/workflows/ci_cd.yml [hunk 2]
```diff
@@ -73,6 +73,8 @@ jobs:
         # if other containers must be tested.
         run:  |
           docker pull ghcr.io/ansys/pydpf-composites:${{ env.CONTAINER_TAG }}
```
Comment (janvonrickenbach) [inline .github/workflows/ci_cd.yml:75]: Add more containers for backward compatibility test

File: src/ansys/dpf/composites/constants.py [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
+    "REF_SURFACE_NAME",
+    "FAILURE_LABEL",
+    "TIME_LABEL",
+)
+
+FAILURE_LABEL = "failure_label"
+TIME_LABEL = "time"
 
 
 class Spot(IntEnum):
```
Comment (roosre) [inline src/ansys/dpf/composites/constants.py:17]: Please document if the indices must be in sync with the backend. Else you can ignore this comment.

**Review by roosre — APPROVED** [review]

File: src/ansys/dpf/composites/constants.py (hunk 3 above)
Comment (janvonrickenbach) [reply src/ansys/dpf/composites/constants.py:17]: @roosre This indices are mapped in _get_rst_spot_index. The mapping there has to be always in sync with how the data is stored in the rst file (which will probably never change). So the enum is basically independent of the backend.

Comment (janvonrickenbach) [reply src/ansys/dpf/composites/constants.py:17]: The failure_label indicies are derived from the field name

**Pull request merged**